indico/web/client/js/react/components/category_navigator/viewmodel.jsx

```diff
@@ -0,0 +1,131 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React, {useEffect, useRef, useState} from 'react';
+import ReactDOM from 'react-dom';
+
+import {getCategoryData, getCategorySearchResults} from './api';
+import * as model from './model';
+
+const loadingSpinnerDelay = 1500;
+
+export function DialogViewModel({trigger, categoryId, view: View, actionView, dialogTitle}) {
+  const [state, setState] = useState(model.init());
+  const dialogRef = useRef();
+  const searchFieldRef = useRef();
+
+  useEffect(() => {
+    trigger.addEventListener('click', openDialog);
+    // eslint-disable-next-line
```

Comment:
> FWIW if those functions do not change during re-renders, it would be better to properly put those in the dependency list.

If you don't change something, putting it in a dependency list just makes React perform unnecessary checks. Not sure why that would be better?

Reply:
I think that's a question for a core react dev, but I guess there's a reason why they have their eslint rule that wants you to put anything you use in the dependency list. Unless it's just because in most cases you want stuff there, and it cannot track whether you really never change certain things...

Reply:
> but I guess there's a reason why they have their eslint rule that wants you to put anything you use in the dependency list

I wouldn't make that assumption myself. 😂